AD/LDAP and SAML updates. (#3213)
mattermost/docs: source/administration/config-settings.rst

| This feature's ``config.json`` setting is ``"UserFilter": ""`` with string input.                                                                                    |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Filter
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
(Optional) Enter an AD/LDAP Filter to use when searching for external users who have Guest Access to Mattermost. Only the objects selected by the query will be accessible to Mattermost.

> This filter is used only when AD/LDAP Group Sync is enabled. See `AD/LDAP Group Sync documentation <https://docs.mattermost.com/deployment/ldap-group-sync.html>`__ for more information

Reviewer: Omit this, this is not accurate.  The filter is used independently of the Group feature.  Potentially we can link the Guest accounts feature doc here so admins can learn more about the feature?

Author: Looks like that's my bad copy and paste job. So the second paragraph should be removed, or both? Happy to provide a link.

Author: Link added.

| This feature's ``config.json`` setting is ``"UserFilter": ""`` with string input.                                                                                    |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Filter
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> (Optional) Enter an AD/LDAP Filter to use when searching for external users who have Guest Access to Mattermost. Only the objects selected by the query will be accessible to Mattermost.

Reviewer 2: The second line isn't correct.  Possibly, add "as Guests." to the end. But other users, can still access Mattermost. Also, should be users, not objects.

Author: Changed it to:
"Optional) Enter an AD/LDAP Filter to use when searching for external users who have Guest Access to Mattermost. Only the users selected by the query will be able to log in to and use Mattermost as Guests."

> (Optional) Enter an AD/LDAP Filter to use when searching for external users who have Guest Access to Mattermost. Only the objects selected by the query will be accessible to Mattermost.

Author: ```suggestion
(Optional) Enter an AD/LDAP Filter to use when searching for external users who have Guest Access to Mattermost. Only the users selected by the query will be accessible to Mattermost as guests.
```
Wondering if "external users" can work as just "users" unless internal/external is defined in AD/LDAP?

See the `Guest Accounts documentation <https://docs.mattermost.com/deployment/guest-accounts.html>`__ for more information
(available in Enterprise Edition E20 and higher).

> This filter is defaulted to ``(objectClass=guests)``.

Reviewer 2: Default is blank or empty.

Author: So we'd say "The filter is defaulted to blank" or "the filter default is blank"? I guess it's because it's optional which I didn't consider - I used the UI text as a guideline so that's good to know. Thanks!

Author: I think I have a better idea of what you mean - the filter default is blank (the UI text is just a guideline) and values have to be explicitly added if being used?

Author: Have edited this.

| This feature's ``config.json`` setting is ``"IdAttribute": ""`` with string input.                                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Attribute
^^^^^^^^^^^^^^^^^

> (Optional) The attribute in the SAML Assertion used to apply a guest role to users in Mattermost.

Reviewer: Potentially we can link the Guest accounts feature doc here so admins can learn more about the feature?

Author: Can definitely do that.

Author: Done.

| This feature's ``config.json`` setting is ``"IdAttribute": ""`` with string input.                                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Attribute
^^^^^^^^^^^^^^^^^

> (Optional) The attribute in the SAML Assertion used to apply a guest role to users in Mattermost.

Reviewer 2: Wondering if we should say "The attribute and value in the SAML...". And potentially provide a sample. 
ie. "IsGuest=true" or "UserType=Guest".

Author: That works for me!

> (Optional) The attribute in the SAML Assertion used to apply a guest role to users in Mattermost.

Author: ```suggestion
(Optional) The attribute and value in the SAML Assertion used to apply a guest role to users in Mattermost (i.e., "IsGuest=true" or "UserType=Guest").
```
Are there any other attribute/value combinations - wondering about the use of i.e., and whether e.g., might be more appropriate.

Reviewer 2: No I think that is only one.

| This feature's ``config.json`` setting is ``"IdAttribute": ""`` with string input.                                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Attribute
^^^^^^^^^^^^^^^^^
(Optional) The attribute in the SAML Assertion used to apply a guest role to users in Mattermost.

> See the `Guest Accounts documentation <https://docs.mattermost.com/deployment/guest-accounts.html>`__ for more information

Author: ```suggestion
See the `Guest Accounts documentation <https://docs.mattermost.com/deployment/guest-accounts.html>`__ for more information.
```

> (available in Enterprise Edition E20 and higher).

Reviewer: Following the format of other config settings, this should be *Available in Enterprise Edition E20* (omit paretheses and remove "and higher"

> (available in Enterprise Edition E20 and higher).

Author: ```suggestion
Available in Enterprise Edition E20.
```

Reviewer: Think this still needs the italics *Available in Enterprise Edition E20.*

| This feature's ``config.json`` setting is ``"IdAttribute": ""`` with string input.                                                                                   |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Guest Attribute
^^^^^^^^^^^^^^^^^
(Optional) The attribute in the SAML Assertion used to apply a guest role to users in Mattermost.

See the `Guest Accounts documentation <https://docs.mattermost.com/deployment/guest-accounts.html>`__ for more information

> Available in Enterprise Edition E20.

Reviewer 3: ```suggestion

*Available in Enterprise Edition E20*
```
